sdk/rust/nym-sdk/src/mixnet/native_client.rs

```diff
@@ -51,6 +56,24 @@ pub struct MixnetClient {
 
     // internal state used for the `Stream` implementation
     _buffered: Vec<ReconstructedMessage>,
+
+    // internal state used for the `AsyncRead` implementation
+    _read: ReadBuffer,
```

Comment:
I think so, since AsyncRead, just uses Stream under the hood, worst case that comes to mind is that you miss a message if you mess up AsyncRead. And since both traits require `&mut` you can't use both at the same exact time.